lib/ApplicationFeatures/ApplicationServer.h

```diff
@@ -127,14 +127,12 @@ class ApplicationServer {
     std::function<void(State, std::string const& featureName)> _feature;
    };
 
-   static ApplicationServer& server();
-
    static std::atomic<bool> CTRL_C;
```

Comment:
eventually we should get rid of this static variable too.

Reply:
Assuming you are referring to `CTRL_C`, I think this one is actually okay. Two reasons.

1) We do indeed want all instances of `ApplicationServer` to share this, and to shut down when receiving `SIGINT`.

2) It seems there's no good way to capture an instance variable to use with the signal handler. The recommended solutions are to either use a static variable like we have done, or to use some singleton class explicitly for the purpose. Given that the functionality is tied to `ApplicationServer`, I'd rather not create a separate singleton class.